Client/game_sa/CModelInfoSA.cpp

```diff
@@ -483,21 +521,278 @@ BOOL CModelInfoSA::DoIsLoaded()
+            return m_pInterface->bAdditiveRender;
+        case eModelIdeFlag::IGNORE_LIGHTING:
+            return m_pInterface->bAdditiveRender;
+        case eModelIdeFlag::NO_ZBUFFER_WRITE:
+            return m_pInterface->bDontWriteZBuffer;
+        case eModelIdeFlag::DONT_RECEIVE_SHADOWS:
+            return m_pInterface->bDontCastShadowsOn;
+        case eModelIdeFlag::IS_GLASS_TYPE_1:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::GLASS_1;
+        case eModelIdeFlag::IS_GLASS_TYPE_2:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::GLASS_2;
+        case eModelIdeFlag::IS_GARAGE_DOOR:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::GARAGE_DOOR;
+        case eModelIdeFlag::IS_TREE:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::TREE;
+        case eModelIdeFlag::IS_PALM:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::PALM;
+        case eModelIdeFlag::IS_TAG:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::TAG;
+        case eModelIdeFlag::IS_BREAKABLE_STATUE:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::BREAKABLE_STATUE;
+        case eModelIdeFlag::IS_CRANE:
+            return m_pInterface->eSpecialModelType == eModelSpecialType::CRANE;
+        case eModelIdeFlag::IS_DAMAGABLE:
+            return ((bool (*)())m_pInterface->VFTBL->AsDamageAtomicModelInfoPtr)();
+        case eModelIdeFlag::DOES_NOT_COLLIDE_WITH_FLYER:
+            return m_pInterface->bDontCollideWithFlyer;
+        case eModelIdeFlag::DISABLE_BACKFACE_CULLING:
+            return !m_pInterface->bIsBackfaceCulled;
+        default:
+            return false;
+    }
+}
+
+void CModelInfoSA::SetModelSpecialType(eModelSpecialType eType, bool bState)
+{
+    if (bState)
+        m_pInterface->eSpecialModelType = eType;
+    else if (m_pInterface->eSpecialModelType == eType)
+        m_pInterface->eSpecialModelType = eModelSpecialType::NONE;
+}
+
+void CModelInfoSA::StaticResetFlags()
+{
+    // Restore default values
+    for (std::map<DWORD, unsigned short>::const_iterator iter = ms_ModelDefaultFlagsMap.begin(); iter != ms_ModelDefaultFlagsMap.end(); ++iter)
+    {
+        CBaseModelInfoSAInterface* pInterface = ppModelInfo[iter->first];
+        if (pInterface)
+        {
+            // Don't change bIsColLoaded flag
+            ushort usFlags = iter->second;
+            usFlags &= 0xFF7F;
+            usFlags |= pInterface->usFlags & 0x80; 
+            pInterface->usFlags = usFlags;
+        }
+    }
+
+    ms_ModelDefaultFlagsMap.clear();
+}
+
+//BYTE CModelInfoSA::GetStreamFlags()
+//{
+//    DWORD dwFunc = FUNC_GetModelFlags;
+//    DWORD ModelID = m_dwModelID;
+//    BYTE  bFlags = 0;
+//    _asm
+//    {
+//        push    ModelID
+//        call    dwFunc
+//        add     esp, 4
+//        mov     bFlags, al
+//    }
+//    return bFlags;
+//}
+
```

Comment:
Left in by accident?

Reply:
Old code for GetFlags function but with wrong behavior